File: automator/src/main/java/com/stardust/automator/filter/StringMatchesFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

/**
 * Created by Stardust on 2017/3/9.
 */

public class StringMatchesFilter extends DfsFilter {

    private final String mRegex;
    private final KeyGetter mKeyGetter;

    StringMatchesFilter(String regex, KeyGetter keyGetter) {
        mRegex = regex;
        mKeyGetter = keyGetter;
    }

    @Override
    protected boolean isIncluded(UiObject nodeInfo) {
        String key = mKeyGetter.getKey(nodeInfo);
        return key != null && key.matches(mRegex);
    }

    @Override
    public String toString() {
        return mKeyGetter.toString() + "Matches(\"" + mRegex + "\")";
    }
}

File: automator/src/main/java/com/stardust/automator/filter/StringMatchesFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

class StringMatchesFilter internal constructor(private val mRegex: String, private val mKeyGetter: KeyGetter) : Filter {

    override fun filter(node: UiObject): Boolean {
        val key = mKeyGetter.getKey(node)
        return key != null && key.matches(mRegex.toRegex())
    }

    override fun toString(): String {
        return mKeyGetter.toString() + "Matches(\"" + mRegex + "\")"
    }
}
